From 1dd22bc13f4b7e4e1823fd4dd5e9772466be2b29 Mon Sep 17 00:00:00 2001
From: kinichiro <kinichiro.inoguchi@gmail.com>
Date: Wed, 1 Jul 2020 17:54:40 +0900
Subject: [PATCH 1552/1936] Check if OS has C++ std::complex, and use this
 condition to include <complex> or not in globals.h, instead of checking OS
 macro

---
 src/CMakeLists.txt | 5 +++++
 src/globals.h      | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 35856391..31535345 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -58,6 +58,7 @@ CHECK_CXX_SOURCE_COMPILES(
 
 check_cxx_compiler_flag("-msse2" SUPPORT_SSE)
 check_cxx_compiler_flag("-mfpu=neon -Werror" SUPPORT_NEON)
+check_include_file_cxx(complex HAVE_CPP_STD_COMPLEX)
 
 set(CMAKE_REQUIRED_FLAGS "")
 
@@ -120,6 +121,10 @@ else()
     add_definitions(-DHAVE_ASYNC=0)
 endif()
 
+if(HAVE_CPP_STD_COMPLEX)
+    add_definitions(-DHAVE_CPP_STD_COMPLEX=1)
+endif()
+
 if(DemoMode)
     add_definitions(-DDEMO_VERSION=1)
 endif()
diff --git a/src/globals.h b/src/globals.h
index cdc0d698..7d95f33b 100644
--- a/src/globals.h
+++ b/src/globals.h
@@ -26,7 +26,7 @@
 
 //Forward Declarations
 
-#if defined(__APPLE__) || defined(__FreeBSD__)
+#if defined(HAVE_CPP_STD_COMPLEX)
 #include <complex>
 #else
 namespace std {
-- 
2.47.0

